File: src/test/resources/dev/hilla/generator/example/BasicEndpoint.ts
import client from '../../../../connect-client.default.js';
import { EndpointRequestInit } from '@hilla/frontend';

const fromArrayOfStringsToNothing = async (argument: string[], init?: EndpointRequestInit): Promise<void> => {
    return client.call('dev.hilla.generator.example.BasicEndpoint', 'fromArrayOfStringsToNothing', { argument }, init);
};

const fromNothingToListOfStrings = async (init?: EndpointRequestInit): Promise<string[]> => {
    return client.call('dev.hilla.generator.example.BasicEndpoint', 'fromNothingToListOfStrings', {}, init);
};

const fromPrimitiveToPrimitiveArray = async (argument: number, init?: EndpointRequestInit): Promise<number[]> => {
    return client.call('dev.hilla.generator.example.BasicEndpoint', 'fromPrimitiveToPrimitiveArray', { argument }, init);
};

const fromStringToString = async (argument: string, init?: EndpointRequestInit): Promise<string> => {
    return client.call('dev.hilla.generator.example.BasicEndpoint', 'fromStringToString', { argument }, init);
};

const fromTwoToMapOfStringsToList = async (init: string, end: string, _init?: EndpointRequestInit): Promise<Map<string, string[]>> => {
    return client.call('dev.hilla.generator.example.BasicEndpoint', 'fromTwoToMapOfStringsToList', { init, end }, _init);
};

const BasicEndpoint = {
    fromArrayOfStringsToNothing,
    fromNothingToListOfStrings,
    fromPrimitiveToPrimitiveArray,
    fromStringToString,
    fromTwoToMapOfStringsToList,
};

export default BasicEndpoint;

File: src/test/java/dev/hilla/generator/example/BasicEndpoint.java
package dev.hilla.generator.example;

import dev.hilla.parser.annotations.Endpoint;
import java.util.List;
import java.util.Map;

@Endpoint
public class BasicEndpoint {
  public String fromStringToString(String argument) {
    return argument;
  }

  public List<String> fromNothingToListOfStrings() {
    return List.of();
  }

  public void fromArrayOfStringsToNothing(String[] argument) {}

  public int[] fromPrimitiveToPrimitiveArray(int argument) {
    return new int[] {argument};
  }

  public Map<String, List<String>> fromTwoToMapOfStringsToList(String init, String end) {
    return Map.of();
  }
}
